lib/db.js

```diff
@@ -4,9 +4,32 @@ const _ = require('lodash/fp')
 const psqlUrl = require('../lib/options').postgresql
 const logger = require('./logger')
 const eventBus = require('./event-bus')
+const extendedQueries = require('./extendedQueries')
 
 const pgp = Pgp({
   pgNative: true,
+  extend (obj, dbContext) {
+    obj.taskEx = function () {
```

Comment:
Are we planning on using taskEx outside of the extendedQueries file? if not we might wanna prepend a underscore to it's name and maybe simplify the logic here.

```js
const getSchema() => {}
const getDefaultSchema() => {}

const searchPathWrapper = (t, cb) => {
    return t.none('SET search_path to $1:name', [schema])
      .then(cb.bind(t, t))
      .finally(t.none('set search_path to $1~', [getDefaultSchema()])))
}

const pgp = Pgp({
  pgNative: true,
  obj._taskEx = myTx = function(cb) {
    const schema = getSchema()
    
    if (!schema) {
      return Promise.reject(new Error('No schema selected, cannot complete query'))
    }
    
    return obj.task.call(this, args.options, t => searchPathWrapper(t, cb))
  },
   obj.query = (query, values, qrm) => obj._taskEx(() => obj.query(query, values, qrm))
}
```

Reply:
This is untested code but I'm pretty sure it would be enough.

I'm also starting to like your idea of not prepending the query names with $, imagine that the query resetting the search_path fails, a query that is not using $ appended version would be able to perform normally and might leak some bugs.

Reply:
Looks like this was not changed .